### PR TITLE
MISUV-7470: Error page - submission validation

### PR DESCRIPTION
**Pull request by james-king-hmrc**
Error page - submission validation

### CODE REVIEW

File: app/controllers/optOut/OptOutErrorController.scala
Comment (tharunprasath) [inline app/controllers/optOut/OptOutErrorController.scala:1]: OptOutErrorController UTs and ITs need to be added.

**Review by tharunprasath — CHANGES_REQUESTED** [review]

File: app/views/optOut/OptOutError.scala.html
Comment (tharunprasath) [inline app/views/optOut/OptOutError.scala.html:1]: Do we agree not to use HTML tags?

Comment (james-king-hmrc) [reply app/views/optOut/OptOutError.scala.html:1]: yes for pages using forms to be submitted, do you think we need Nunjucks here?

Comment (tharunprasath) [reply app/views/optOut/OptOutError.scala.html:1]: Reuse ErrorTemplate.scala.html & ErrorTemplateWithLink.scala.html

File: app/controllers/optOut/OptOutErrorController.scala
Comment (james-king-hmrc) [reply app/controllers/optOut/OptOutErrorController.scala:1]: done

File: app/views/optOut/OptOutError.scala.html
Comment (james-king-hmrc) [reply app/views/optOut/OptOutError.scala.html:1]: not sure they support the the format we need here

File: app/controllers/optOut/OptOutErrorController.scala
Comment (tharunprasath) [reply app/controllers/optOut/OptOutErrorController.scala:1]: We could add the logger to print the reasons for API failure. All the API failures are generalised under OptOutUpdateResponseFailure.defaultFailure. We could add the possible test data for the listed failure codes in our test as error scenarios won't be covered in ATs.

File: app/views/optOut/OptOutError.scala.html
Comment (james-king-hmrc) [reply app/views/optOut/OptOutError.scala.html:1]: done

File: app/controllers/optOut/OptOutErrorController.scala
Comment (james-king-hmrc) [reply app/controllers/optOut/OptOutErrorController.scala:1]: not sure I understand this comment, let's chat when you're free to calrify

**Review by SiddharthAtos — APPROVED** [review]

File: app/controllers/optOut/OptOutErrorController.scala
Comment (james-king-hmrc) [reply app/controllers/optOut/OptOutErrorController.scala:1]: this comment has been addressed and apporved, please reslove

File: app/views/optOut/OptOutError.scala.html
Comment (james-king-hmrc) [reply app/views/optOut/OptOutError.scala.html:1]: this comment has been addressed and apporved, please reslove

File: it/test/controllers/optOut/ConfirmOptOutControllerISpec.scala [hunk 1]
```diff
@@ -166,8 +165,7 @@ class ConfirmOptOutControllerISpec extends ComponentSpecBase {
         val result = IncomeTaxViewChangeFrontendManageBusinesses.postConfirmOptOut()
 
         result should have(
-          httpStatus(SEE_OTHER),
-          redirectURI(optOutErrorPageUrl)
+          httpStatus(SEE_OTHER)
```
Comment (SiddharthAtos) [inline it/test/controllers/optOut/ConfirmOptOutControllerISpec.scala:168]: should fix: redirectURI should be verified  with SEE_OTHER status

**Pull request merged**